### ISSUE TITLE
Wrong SC2155 check for local variables

### ISSUE THREAD

**Issue by BorzdeG** [koalaman/shellcheck#1089]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2155
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [ x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [  ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash

function demo0(){
        echo 'test'
}

function demo(){
        local var0=$(demo0)
        var0="${var0}add"
        echo "${var0}"
}

demo
echo "$var0"

```

#### Here's what shellcheck currently says:

```
Line 8:
        local var0=$(demo0)
              ^-- SC2155: Declare and assign separately to avoid masking return values.
```

#### Here's what I wanted or expected to see:

No errors

**Comment by Nightfirecat** [comment]
That looks right to me. `$?` will always be `0` with that statement, which is what the warning is there for.

**Comment by koalaman** [comment]
The exit code of `demo0` is masked is definitely being masked by `local`, even if this script by itself doesn't care, so it seems to be working as intended
